### PR TITLE
Migrated Textbox edit dialog to React

### PR DESCRIPTION
**Pull request by ranbena**
## What type of PR is this?

- [x] Refactor

## Description
Renamed `AddTextboxDialog` to `TextboxDialog` and adjusted to accept `this.props.text`.
Notice there are already e2e tests that verify textbox add and edit 🏋

## Mobile & Desktop Screenshots/Recordings
Before:

After:

### CODE REVIEW

**Comment by ranbena** [comment]
@kravets-levko can you give this a quick look?
I split it into 2 commits for review convenience.

File: client/app/components/dashboards/TextboxDialog.jsx [hunk 1]
```diff
@@ -53,20 +62,21 @@ class AddTextboxDialog extends React.Component {
...
         okButtonProps={{
           loading: this.state.saveInProgress,
           disabled: !this.state.text,
         }}
-        okText="Add to Dashboard"
+        okText={isNew ? 'Add to Dashboard' : 'Save'}
         width={500}
       >
-        <div className="add-textbox">
+        <div className="textbox-dialog">
```
Comment (kravets-levko) [inline client/app/components/dashboards/TextboxDialog.jsx]: Need to update classname in `TextboxDialog.less` as well

Comment (ranbena) [reply client/app/components/dashboards/TextboxDialog.jsx]: 10x I missed that 😓 Fixed.

File: client/cypress/integration/dashboard/dashboard_spec.js [hunk 2]
```diff
@@ -256,15 +256,15 @@ describe('Dashboard', () => {
           });
 
         const newContent = '[edited]';
-        cy.get('edit-text-box')
+        cy.get('.textbox-dialog')
```
Comment (kravets-levko) [inline client/cypress/integration/dashboard/dashboard_spec.js]: You can add `wrapProps={{ 'data-test': 'TextboxDialog' }}` to the dialog and then use it in tests (https://github.com/getredash/redash/pull/3493/files#diff-0c11d795d0fc2f8315e6f3a946848ea7R193)

Comment (ranbena) [reply client/cypress/integration/dashboard/dashboard_spec.js]: Oooh I'll do that.

**Pull request merged**